File: VisualBasic/Sample/ArraySample.vb
Public Class ArraySample

    Shared Sub test()

        '/////////////// 配列 ///////////////

        '生成
        Dim ary() As String = {"Honda", "Yamaha", "Kawasaki"}
        Debug.WriteLine(ary(0))
        Debug.WriteLine(ary(1))
        Debug.WriteLine(ary(2))

        Dim ary2(3) As String
        ary2(0) = "Honda"
        ary2(1) = "Yamaha"
        ary2(2) = "Kawasaki"


        '走査
        For Each v As String In ary
            Debug.WriteLine(v)
        Next

        For i = 0 To ary.Count - 1
            Debug.WriteLine(ary(i))
        Next

        For i = 0 To UBound(ary)
            Debug.WriteLine(ary(i))
        Next

        '存在確認
        If ary.Contains("Honda") Then
            Debug.WriteLine("存在確認")
        End If
        If Array.IndexOf(ary, "Yamaha") = 1 Then
            Debug.WriteLine("Yamaha index = 1")
        End If

        '要素数の変更
        ReDim Preserve ary(5) 'Preserveをつけないと今までの値が保持されない
        ary(3) = "Suzuki"
        For Each v As String In ary
            Debug.WriteLine(v)
        Next

        '要素数
        Debug.WriteLine(ary.Length)

        'ソート
        Dim intAry() As Integer = {7, 100, 22, 99, 51, 0}
        Array.Sort(intAry)
        For Each v As String In intAry
            Debug.WriteLine(v)
        Next


        '/////////////// リスト ///////////////

        Dim myList As New List(Of String)(New String() {"Honda", "Yamaha", "Kawasaki"}) 'ジェネリック

        Dim myList2 As New List(Of String)
        '要素の追加
        myList2.Add("Honda")
        myList2.Add("Yamaha")
        myList2.Add("Kawasaki")

        '書き換え
        myList2(1) = "Harley"

        '要素の削除
        myList2.RemoveAt(1)


        '走査
        For i = 0 To myList.Count - 1
            Debug.WriteLine(myList(i)) ' 赤 黄 青が出力される
        Next i

        For Each s In myList
            Debug.WriteLine(s)
        Next

        'インデックス
        Debug.WriteLine(myList.IndexOf("Yamaha"))

        '存在確認
        If myList.Contains("Yamaha") Then
            Debug.WriteLine("存在確認")
        End If

        '要素数
        Debug.WriteLine(myList.Count)

        'リストから配列にコピー
        Dim myList3 = myList.ToArray()



        '/////////////// 辞書 ///////////////

        '生成
        Dim dict As New Dictionary(Of String, Integer) From {
            {"Japan", 1},
            {"America", 2},
            {"Germany", 3}
            }

        'アクセス
        Debug.WriteLine(dict("Germany"))

        '追加
        dict.Add("France", 10)
        dict("France") = 1000

        '削除
        dict.Remove("France")

        '走査
        For Each item In dict
            Debug.WriteLine(item.Key)
            Debug.WriteLine(item.Value)
        Next
        For i = 0 To dict.Count - 1
            Debug.WriteLine(dict.Keys(i))
            Debug.WriteLine(dict.Values(i))
        Next

        '存在確認
        If (dict.ContainsKey("Germany")) Then
            Debug.WriteLine("contains")
        End If

        If (dict.ContainsValue(2)) Then
            Debug.WriteLine("contains")
        End If

        '要素数
        Debug.WriteLine(dict.Count)


    End Sub
End Class

File: C#/ArraySample.cs
using System;
using System.Collections.Generic;

class ArraySample {

    public ArraySample() {

        ////////////// 配列 //////////////

        // 初期化
        int[] ary = new int[6] { 0, 1, 2, 3, 4, 5 };

        // 変更
        ary[5] = 100;

        // 要素数
        Console.WriteLine(ary.Length);

        // リサイズ（末尾に1要素を追加）
        Array.Resize(ref ary, ary.Length + 1);
        ary[ary.Length - 1] = 1000;

        // 走査
        for (int i = 0; i < ary.Length; i++) {
            Console.WriteLine("{0}: {1}", i + 1, ary[i]);
        }
        foreach (int i in ary) {
            Console.WriteLine(i);
        }
        Console.WriteLine("---");

        // コピー

        // Shallow
        int[] ary2 = ary;
        ary[0] = 99;
        foreach (int i in ary2) {
            Console.WriteLine(i); // aryの変更がary2に影響している
        }
        Console.WriteLine("---");

        // Deep
        int[] ary3 = new int[ary.Length];
        Array.Copy(ary, ary3, ary.Length);
        ary[0] = 9999;
        foreach (int i in ary3) {
            Console.WriteLine(i); // aryの変更がary3に影響しない
        }
        Console.WriteLine("---");

        // ソート
        ary = new int[6] {100, 1, 6, 9, 0, 7};
        Array.Sort(ary);
        foreach (int i in ary) {
            Console.WriteLine(i);
        }


        ////////////// リスト //////////////

        // 生成
        var list = new List<string>() {"Honda", "Yamaha"};

        // 変更
        list[1] = "Kawasaki";

        // 削除
        list.RemoveAt(0); // indexを指定
        list.Remove("Kawasaki"); // 要素を指定

        // 末尾に追加
        list.Add("Honda");
        list.Add("Yamaha");

        // 位置を指定して追加
        list.Insert(0, "Harley");

        // 要素数
        Console.WriteLine(list.Count);

        // 走査
        for (int i = 0; i < list.Count; i++) {
            Console.WriteLine("{0}: {1}", i + 1, list[i]);
        }
        foreach (string s in list) {
            Console.WriteLine(s);
        }

        // index取得
        Console.WriteLine(list.IndexOf("Honda"));

        // 存在確認
        Console.WriteLine(list.Contains("Harley"));


        ////////////// 辞書 //////////////

        // 生成
        var dict = new Dictionary<string, int>()
        { {"a", 100},
          {"b", 101},
          {"c", 102}
        };

        // 変更
        dict["a"] = 0;

        // 追加
        dict.Add("d", 103);

        // 削除
        dict.Remove("d");

        // 要素数
        Console.WriteLine(dict.Count);

        // 走査
        foreach (KeyValuePair<string, int> pair in dict) {
            Console.WriteLine("{0}-{1}", pair.Key, pair.Value);
        }
        foreach (string key in dict.Keys) {
            Console.WriteLine(key);
        }
        foreach (int val in dict.Values) {
            Console.WriteLine(val);
        }

        // 存在確認
        Console.WriteLine(dict.ContainsKey("a"));
        Console.WriteLine(dict.ContainsValue(102));


        ////////////// セット型 //////////////

        var hs = new HashSet<string>() {
            "A", "B", "C", "A"
        };

        hs.Add("B");
        hs.Remove("B");

        foreach (var str in hs) {
            Console.WriteLine(str);
        }
        
    }
}